ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -456,6 +486,13 @@ private void initialize(final KsqlConfig configWithApplicationServer) {
   @SuppressWarnings("checkstyle:NPathComplexity")
   @Override
   public void triggerShutdown() {
+    // First, make sure the server wasn't stuck in startup.  Set the shutdown flag and interrupt the
+    // startup thread if it's been hanging.
+    shuttingDown.set(true);
+    if (startAsyncThread != null) {
+      startAsyncThread.interrupt();
```

Comment:
If we're interrupting the start up thread, then there's a high chance we'll see interrupt exceptions being thrown. Yet I don't see anywhere that we catch this, now expected, exception.  This may lead to the exception being logged out, which isn't ideal if its expected.

Maybe the code in `startAsync` need to also catch interrupted exceptions and log out the `Aborting application start`?

Reply:
In this particular case, it's caught in `RetryUtil`:

```
duration -> {
          try {
            Thread.sleep(duration);
          } catch (final InterruptedException e) {
            log.debug("retryWithBackoff interrupted while sleeping");
          }
        },
```

I believe it would only throw this exception if someone did something that could be interrupted (like sleeping), and since it's a checked exception, they would have to catch it explicitly (or let it bubble up explicitly).  In this case, I would want them to catch it and then throw a `AbortApplicationStartException` which is already handled.  So I don't think catching `InterruptedException`s at the `startAsync` level would be required.  Is that reasonable?